### PR TITLE
pkg: enhance packages doxygen documentation

### PR DESCRIPTION
**Pull request by aabadie**
I realized that the package documentation was not complete:
* some important new packages being not documented: openthread, spiffs
* packages not referenced in global groups such as net or sys
* some packages documented with README and some for doxygen

This PR addresses these problems for the most important ones. I replaced the README with doxygen documentation when needed.

### CODE REVIEW

File: pkg/fatfs/doc.txt [hunk 1]
```diff
@@ -0,0 +1,7 @@
+/**
+ * @defgroup pkg_fatfs   FAT file system
+ * @ingroup  pkg
+ * @ingroup  sys
```
Comment (miri64) [inline pkg/fatfs/doc.txt]: `sys_fs`?

File: pkg/libfixmath/doc.txt [hunk 2]
```diff
@@ -0,0 +1,7 @@
+/**
+ * @defgroup pkg_lib_fix_math Cross platform fixed point maths library
```
Comment (miri64) [inline pkg/libfixmath/doc.txt]: `pkg_libfixmath`?

File: pkg/micro-ecc/doc.txt [hunk 3]
```diff
@@ -0,0 +1,46 @@
+/**
+ * @defgroup pkg_micro_ecc   Micro-ECC for RIOT
+ * @ingroup  pkg
+ * @ingroup  sys
```
Comment (miri64) [inline pkg/micro-ecc/doc.txt]: `sys_crypto`?

File: pkg/relic/doc.txt [hunk 4]
```diff
@@ -0,0 +1,28 @@
+/**
+ * @defgroup pkg_relic  Relic toolkit for RIOT
+ * @ingroup  pkg
+ * @ingroup  sys
```
Comment (miri64) [inline pkg/relic/doc.txt]: `sys_crypto`?

File: pkg/spiffs/doc.txt [hunk 5]
```diff
@@ -0,0 +1,7 @@
+/**
+ * @defgroup pkg_spiffs   SPI flash file system
+ * @ingroup  pkg
+ * @ingroup  sys
```
Comment (miri64) [inline pkg/spiffs/doc.txt]: `sys_fs`?

File: pkg/tiny-asn1/doc.txt [hunk 6]
```diff
@@ -1,6 +1,7 @@
 /**
- * @defgroup tiny-asn1 tiny-asn1
- * @ingroup pkg
- * @brief   Lightweight ASN.1 decoding/encoding library
- * @see     https://gitlab.com/matthegap/tiny-asn1
+ * @defgroup pkg_tiny-asn1 Lightweight ASN.1 decoding/encoding library
+ * @ingroup  pkg
+ * @ingroup  sys
```
Comment (miri64) [inline pkg/tiny-asn1/doc.txt]: `sys_crypto`?

File: pkg/wakaama/doc.txt [hunk 7]
```diff
@@ -0,0 +1,6 @@
+/**
+ * @defgroup pkg_wakaama  Wakaama LwM2M implementation
+ * @ingroup  pkg
```
Comment (miri64) [inline pkg/wakaama/doc.txt]: also add to `net`?

File: pkg/ccn-lite/ccn-lite-riot.h [hunk 8]
```diff
@@ -12,6 +12,7 @@
 /**
  * @defgroup    pkg_ccnlite CCN-Lite stack
  * @ingroup     pkg
+ * @ingroup     net
```
Comment (smlng) [inline pkg/ccn-lite/ccn-lite-riot.h]: IMHO adding a second group here is not a good idea, these are third party, external packages, adding additional grouping like `net`, `sys`, `sys_fs` or any other RIOT core grouping will mix things up in the documentation.

I know I introduces this _features_ of multiple groups for `cpu/periph/` drivers, but these are RIOT internal, hence that fits.

Comment (smlng) [inline pkg/ccn-lite/ccn-lite-riot.h]: this is more a comment, than an actual change request. I will not block this effort 'cause improving documentation is always good 😉 

@miri64 what is your opinion on those additional groups?

Comment (aabadie) [inline pkg/ccn-lite/ccn-lite-riot.h]: From the documentation point of view, multi groups for pkg makes sense to me: are Openthread, lwIP, emb6 network packages ? Yes, so they should appear in Networking root group and package group.
Otherwise, we can create subgroups of pkgs but it will become a mess.

Comment (miri64) [inline pkg/ccn-lite/ccn-lite-riot.h]: I agree with @aabadie here (which was, why I did not comment on that in the first place).

Comment (miri64) [inline pkg/ccn-lite/ccn-lite-riot.h]: but maybe this is a discussion we should have on `devel@` ;-)

**Comment by aabadie** [comment]
@miri64, your comment are addressed. There's just the comment about double grouping from @smlng that remains. Can we just ACK and merge, since the global result is already better than before ?

**Comment by smlng** [comment]
I reviewed the changes also together with #7302, I think its fine to have double groupings. Though there seem to be some artifacts in the doxygen linking - which has nothing to do with these PRs, I think. 

TL;DR: ACK
